### DISCUSSION TITLE
I have two similar tables with similar access, both I can read by CURL, but with @supabase/supabase-js I can read only one table

### DISCUSSION THREAD

**Discussion by AAlex-11** [supabase/supabase#30490] [Questions] [answered]
```
import { createClient } from "@supabase/supabase-js";
import {} from "dotenv/config";

const supabaseUrl = process.env.SUPABASE_URL;
const supabaseKey = process.env.SUPABASE_KEY;
const supabase = createClient(supabaseUrl, supabaseKey);

let { data, error } = await supabase.from("types").select();
console.log(data, error)

const supabase1 = createClient(supabaseUrl, supabaseKey);

let{ data1, error1 } = await supabase1.from("entry").select();
console.log(data1, error1)
```
![VbNetSupport-11152024_131144_1](https://github.com/user-attachments/assets/0848b246-84b1-47f7-a05f-76893bcf2e07)

![VbNetSupport-11152024_134500](https://github.com/user-attachments/assets/c06065df-47d2-4cd8-a582-c8bac5fb77af)

![VbNetSupport-11152024_134547](https://github.com/user-attachments/assets/5c6daef6-92f7-4113-b31c-040f5939a89e)

**Comment by GaryAustin1** [comment]
You don't mention your RLS being enabled or not and policies.  
Also what authorization header are you using for CURL versus supabase-js.

**Reply by GaryAustin1** [reply, marked as answer]
Your 2nd request does not have the correct syntax.
You can't just change the name of data and error like that.
You need to use data:data1 and error:error1 to provide an alias.

Also when you respond on discussions you should reply in the same thread and not start a new answer.

**Reply by AAlex-11** [reply]
yes, sorry, you are right, this is my mistake !!!

**Comment by AAlex-11** [comment]
Both table has the same security - permit to readonly for everybody.
![VbNetSupport-11152024_165658_1](https://github.com/user-attachments/assets/1d11f17a-8a1d-45b2-a39e-3186a0190088)
![VbNetSupport-11152024_165740_1](https://github.com/user-attachments/assets/6d0a6839-6262-4548-af9c-2c4c7483613c)
and request to both tables working in the same anonymous way with CURL or any other services like Javascript FETCH or any other rest CLIENT

**ONLY**  "@supabase/supabase-js" allow me access to one table and forbid me access to another table. This is the issue.
I try to write ticket for support to investigate more details, but support have no interest to this issue.

![VbNetSupport-11152024_133832_1](https://github.com/user-attachments/assets/12cfd21a-0a7e-486e-bf15-5b49363e3107)
